include/swift/SILOptimizer/Utils/Differentiation.h

```diff
@@ -35,6 +37,37 @@ using llvm::SmallDenseSet;
 using llvm::SmallMapVector;
 using llvm::SmallSet;
 
+class ApplyInst;
+
+//===----------------------------------------------------------------------===//
+// Helpers
+//===----------------------------------------------------------------------===//
+
+namespace autodiff {
+
+/// Prints an "[AD] " prefix to `llvm::dbgs()` and returns the debug stream.
+/// This is being used to print short debug messages within the AD pass.
+raw_ostream &getADDebugStream();
```

Comment:
Where is the implementation of these functions? When we define a `Utils/Differentiation.h`, we should make sure to have a `Utils/Differentiation.cpp` that implements all functions the header file declares.

Reply:
They were in `SILOptimizer/Mandatory/Differentiation.cpp`. I created a`Utils/Differentiation.cpp`.